copier/copier.go

```diff
@@ -1794,9 +1795,10 @@ func copierHandlerPut(bulkReader io.Reader, req request, idMappings *idtools.IDM
 				}
 			case tar.TypeDir:
 				if err = os.Mkdir(path, 0700); err != nil && os.IsExist(err) {
-					var st os.FileInfo
-					if st, err = os.Lstat(path); err == nil && !st.IsDir() {
-						// it's not a directory, so remove it and mkdir
+					if st, stErr := os.Lstat(path); stErr == nil && !st.IsDir() {
+						if req.PutOptions.NoOverwriteNonDirDir {
+							return errors.Wrapf(err, "copier: put: error creating directory at %q", path)
+						}
```

Comment:
I'd kind of prefer if this followed the style of the various `NoOverwriteDirNonDir` cases in this loop, which break out instead of returning directly, in case we need to add cleanup logic later, but it's not a blocker.

Reply:
Will do, thank you!